libs/wingc/src/type_check.rs

```diff
@@ -4728,20 +4852,60 @@ impl<'a> TypeChecker<'a> {
 		}
 	}
 
-	/// Get's the type of an instance variable in a class
+	/// Get's the type of an instance variable in a class and the type in which it's defined
 	fn get_property_from_class_like(
 		&mut self,
 		class: &impl ClassLike,
 		property: &Symbol,
 		allow_static: bool,
+		env: &SymbolEnv,
 	) -> VariableInfo {
 		let lookup_res = class.get_env().lookup_ext(property, None);
-		if let LookupResult::Found(field, _) = lookup_res {
+		if let LookupResult::Found(field, lookup_info) = lookup_res {
 			let var = field.as_variable().expect("Expected property to be a variable");
+
+			// Determine the access type of the property
+			// Lookup the property in the class env to find out in which class (perhaps an ancestor) it was defined
+			let SymbolEnvKind::Type(property_defined_in) = lookup_info.env.kind else {
+				panic!("Expected env to be a type env");
+			};
+			// Check if the class in which the property is defined is the current class nesting
```

Comment:
What does this sentence mean?

Reply:
I changed to:
"Check if the class in which the property is defined is one of the classes we're currently nested in"
Hopefully that's clearer. I definitely struggled to correctly describe what I'm doing here (and I'm also not happy with the method name `self.ctx.current_class_nesting` 3 lines below).
What I'm doing here is I run up the stack of currently nested classes to see if the property access is from one of those classes. Here's an example to make it clearer:
```wing
class A {
  static foo() {} // This is a **private** static method.
  method() {
    class B {
        method() {
          A.foo() // I can access A's private stuff since I'm nested inside B which is nested inside A.
        }
    }
  }
}
```

Reply:
Got it, yeah makes more sense now  - thanks!